app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -1286,9 +1288,27 @@ public void onBackPressed() {
             getSupportFragmentManager().popBackStack(NARROW,
                     FragmentManager.POP_BACK_STACK_INCLUSIVE);
         } else {
-            super.onBackPressed();
+            if(atHome){
```

Comment:
You can remove the atHome boolean and use narrowedList to check whether it is narrowed to any private/stream messages.
If narrowedList is null it means it is in the homeView.